src-tauri/src/mail_sender.rs

```diff
@@ -0,0 +1,145 @@
+use std::fs;
+use std::path::{Path, PathBuf};
+
+use lettre::message::Mailbox;
+use lettre::message::{Attachment, Body, MultiPart, header::ContentType};
+use lettre::{Address, Message, SmtpTransport, Transport};
+
+use crate::config::config;
+
+use thiserror::Error;
+
+use anyhow::Result;
+
+#[derive(Error, Debug)]
+pub enum MailSenderError {
+    #[error("invalid email")]
+    InvalidEmail,
+
+    #[error("not in database")]
+    NotInDatabase,
+
+    #[error("invalid file path")]
+    InvalidFilePath,
+
+    #[error("no receivers")]
+    NoReceivers,
+
+    #[error("no file")]
+    NoFile,
+
+    #[error("couldn't send email")]
+    CouldntSendEmail,
+}
+
+#[derive(Debug, PartialEq)]
+pub struct Receiver {
+    pub name: String,
+    pub mail: Address,
+}
+
+pub struct MailSender {
+    people: Vec<Receiver>,
+    file_path: Option<PathBuf>,
```

Comment:
proč option? Bude někdy situace, kdy tohle bude prázdný?

Reply:
Hmm, nebude, jak koukám. Imo by to neměl bejt Option, a tim pádem by MailSender neměl implementovat Default

Reply:
V tohle případě bys měl mít MailSender::new(file_path) asociovanou funkci